chariot-server/src/game/phase.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+use std::time::Instant;
+
+use chariot_core::networking::ws::QuestionBody;
+use chariot_core::networking::Uuid;
+
+use crate::chairs::get_player_start_physics_properties;
+
+use super::ServerGameState;
+
+/*
+ * Phases of the game are as follows:
+*
+ * 1. GamePhase::WaitingForPlayerReady
+ *  During this phase, players are in selection screens choosing their chair
+ *  and marking ready; they'll be waiting in UI for all of this phase. The
+ *  transition from Phase 1 to 2 occurs when the server sends all clients an
+ *  GameStart packet, which will include a time to transition from phase 2
+ *  to 3.
+ *
+ * 2. GamePhase::CountingDownToGameStart
+ *  During this phase, players can see the track, the other players, and have
+ *  a countdown until they'll be able to use controls. The transition from Phase
+ *  2 to 3 is timed by the server, synchronized by an earlier GameStart packet.
+ *
+ * 3. GamePhase::PlayingBeforeVoting
+ *  The idea we had was to have idk 30 seconds at the start of the race
+ *  before voting starts; players will be driving around normally and the
+ *  clients will be oblivious to the difference between Phase 3 and 4. (client
+ *  will time itself internally)
+ *
+ * 4. GamePhase::PlayingWithVoting
+ *  All the good stuff happening at once! >:))) The transition from Phase 4 to
+ *  Phase 5 is marked by the server's AllDone packet.
```

Comment:
I think we should have phase 'PlayingWhileVoting` and `PlayingWithDecision`, where the first stage is while players play with no effect and the second stage is players playing with the effect added. These two phases don't make enough sense to be separated to me + i feel are a bit confusing. 

```suggestion
 * 3. GamePhase::PlayingWhileVoting
 *     Players will be driving around normally and the clients will simply 
 *     display "The Audience is Deciding Your Fate" (for juice). 
 *     State: `timeUntilVotingTally`, `questionPrompt`
 *
 * 4. GamePhase::PlayingWithDecision
 *     Players will be doing their thing WITH some new effect in place. 
  *    This effect only exists during this phase. 
 *     State: `timeUntilDecisionEnds`, `winningEffect` (this would be 
  *    an enum or something so we can 
 *     (a) display it on clients and (b) do special server logic)
 *     The transition from Phase 4 to Phase 5 is marked by the server's
 *     AllDone packet (sent when someone wins the game)
```